### ISSUE TITLE
`no-unreachable-types` misses directive on `root schema` object [apollo federation]

### ISSUE THREAD

**Issue by dnalborczyk** [dotansimha/graphql-eslint#748]
**Describe the bug**

**To Reproduce**

the following schema is a subset of an apollo federation generated schema.
`no-unreachable-types` misses the `@core directive` use on the root schema object.
I assume not associating the `core__Purpose` enum is a result of that issue?

```graphql
schema
  @core(feature: "https://specs.apollo.dev/core/v0.2")
  @core(feature: "https://specs.apollo.dev/join/v0.1", for: EXECUTION)
  @core(feature: "https://specs.apollo.dev/tag/v0.1") {
  query: Query
}

directive @core(
  as: String
  feature: String!
  for: core__Purpose
) repeatable on SCHEMA

enum core__Purpose {
  """
  `EXECUTION` features provide metadata necessary to for operation execution.
  """
  EXECUTION

  """
  `SECURITY` features provide metadata necessary to securely resolve fields.
  """
  SECURITY
}
```

```shell
8:11   error  Type "core" is unreachable           @graphql-eslint/no-unreachable-types
78:6   error  Type "core__Purpose" is unreachable  @graphql-eslint/no-unreachable-types
```

**Expected behavior**

**Environment:**

- OS: macOS
- `@graphql-eslint`: v2.3.2
- NodeJS: v17.0.1

**Additional context**

**Comment by B2o5T** [comment]
@dnalborczyk thank you for the report! This will be fixed on the next minor release, the related test was added also

**Comment by dnalborczyk** [comment]
awesome, thank you @B2o5T !

**Comment by dotansimha** [comment]
Fixed in @graphql-eslint/eslint-plugin@2.5.0
